scripts/apps/users/directives/UserPreferencesDirective.ts

```diff
@@ -56,7 +54,53 @@ export function UserPreferencesDirective(
              */
 
             scope.preferencesLoaded = false;
-            var orig; // original preferences, before any changes
+            var orig: {[key: string]: any}; // original preferences, before any changes
+
+            scope.extensionsNotifications = {};
+
+            for (const extension of Object.values(extensions)) {
+                for (const [key, value] of Object.entries(extension.activationResult.contributions?.notifications ?? [])) {
+                    if (value.type === 'email') {
+                        preferencesService.registerUserPreference(key, 1);
+                        scope.extensionsNotifications[key] = preferencesService.getSync(key);
+                    }
+                }
+            }
+
+            scope.toggleEmailGroupNotifications = function() {
+                const isGroupEnabled = scope.preferences['email:notification'].enabled;
+
+                Object.keys(scope.extensionsNotifications).forEach((notificationId) => {
+                    scope.preferences[notificationId].enabled = isGroupEnabled;
+                    scope.extensionsNotifications[notificationId] = {
+                        ...scope.extensionsNotifications[notificationId],
+                        enabled: isGroupEnabled,
+                    };
+                });
+
+                scope.userPrefs.$setDirty();
+                scope.$applyAsync();
+            };
+
+            scope.toggleEmailNotification = function(notificationId: string) {
+                scope.preferences[notificationId] = {
+                    ...(scope.preferences[notificationId] ?? {}),
+                    enabled: !(scope.preferences[notificationId]?.enabled ?? false),
+                };
+
+                scope.extensionsNotifications[notificationId] = {
+                    ...scope.extensionsNotifications[notificationId],
+                    enabled: scope.preferences[notificationId].enabled,
+                };
+
+                const notificationsForGroupAreOff = Object.keys(scope.extensionsNotifications)
```

Comment:
`.some` and `.every` are functionally the same, just reverse. In terms of performance it is the same. Which to use depends on which term more accuratelly describes what you're doing. In this case it seems you're inverting the logic - `const allFalse = arr.some(x => x === true)`. While technically correct, the following matches naming better: `const allFalse = arr.every(x => x === false)`